### PR TITLE
[JENKINS-34464] Bump structs version, add test for result roundtrip

### PR DESCRIPTION
**Pull request by abayer**
[JENKINS-34464](https://issues.jenkins-ci.org/browse/JENKINS-34464)

Downstream of https://github.com/jenkinsci/structs-plugin/pull/19, upstream of https://github.com/jenkinsci/workflow-multibranch-plugin/pull/63

The `workflow-multibranch` test was failing, seemingly due to `workflow-cps` having the older `structs` version (plus some other problems, but those are fixed over in that PR anyway). So let's bump the `structs` release.

cc @reviewbybees

### CODE REVIEW

File: src/test/java/org/jenkinsci/plugins/workflow/testMetaStep/EchoResultStep.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    public static class EchoResultStepExecution extends StepExecution {
+        private final EchoResultStep step;
+
+        public EchoResultStepExecution(EchoResultStep s, StepContext context) {
+            super(context);
+            this.step = s;
+        }
+
+        @Override
+        public boolean start() throws Exception {
```
Comment (jglick) [inline src/test/java/org/jenkinsci/plugins/workflow/testMetaStep/EchoResultStep.java]: Note that the test is not actually running it.

File: src/test/java/org/jenkinsci/plugins/workflow/testMetaStep/EchoResultStep.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+            TaskListener listener = getContext().get(TaskListener.class);
+
+            if (listener != null)
```
Comment (jglick) [inline src/test/java/org/jenkinsci/plugins/workflow/testMetaStep/EchoResultStep.java]: No null check needed, it is required.

Comment (abayer) [inline src/test/java/org/jenkinsci/plugins/workflow/testMetaStep/EchoResultStep.java]: My IDE complained about a possible NPE without it, so hey, I put in the check.

File: src/test/java/org/jenkinsci/plugins/workflow/testMetaStep/EchoResultStep.java (hunk 1 above)
Comment (abayer) [inline src/test/java/org/jenkinsci/plugins/workflow/testMetaStep/EchoResultStep.java]: Yup.

File: src/test/java/org/jenkinsci/plugins/workflow/testMetaStep/EchoResultStep.java (hunk 2 above)
Comment (jglick) [inline src/test/java/org/jenkinsci/plugins/workflow/testMetaStep/EchoResultStep.java]: What is wrong with your IDE? `StepContext.get` is correctly marked `@Nullable`.

**Comment by abayer** [comment]
`InjectedTest` failure is solved in #135. Trying to figure out why the `resultRoundTrips` test is failing, 'cos it isn't locally so far...

File: src/test/java/org/jenkinsci/plugins/workflow/testMetaStep/EchoResultStep.java (hunk 2 above)
Comment (abayer) [inline src/test/java/org/jenkinsci/plugins/workflow/testMetaStep/EchoResultStep.java]: It's complaining at the `listener.getLogger()...`.

**Comment by abayer** [comment]
Ok, I am mystified as to why `SnippetizerTest#resultRoundTrips` is failing here when it works for me locally.

**Comment by jglick** [comment]
```
org.junit.ComparisonFailure: expected:<echoResult ['UNSTABLE']> but was:<echoResult [<object of type hudson.model.Result>]>
	at org.junit.Assert.assertEquals(Assert.java:115)
	at org.junit.Assert.assertEquals(Assert.java:144)
	at org.jenkinsci.plugins.workflow.cps.SnippetizerTester.assertRoundTrip(SnippetizerTester.java:95)
	at org.jenkinsci.plugins.workflow.cps.SnippetizerTest.resultRoundTrips(SnippetizerTest.java:315)
```

specifically. Perhaps you are picking up the wrong snapshot?

**Comment by abayer** [comment]
It's got the right `SNAPSHOT` in the POM - I'm wondering if it's somehow getting messed up and picking up https://repo.jenkins-ci.org/snapshots/org/jenkins-ci/plugins/structs/1.7-SNAPSHOT/structs-1.7-20170522.183630-2.hpi, a SNAPSHOT you deployed a couple hours after https://repo.jenkins-ci.org/snapshots/org/jenkins-ci/plugins/structs/1.7-SNAPSHOT/structs-1.7-20170522.160448-1.hpi, the one in the POM here?

**Comment by jglick** [comment]
(We really need to switch to doing everything inside Docker containers, especially now that we have a repository mirror inside Azure so it should not be too expensive to require every build to download everything.)

File: pom.xml [hunk 3]
```diff
@@ -103,7 +103,7 @@
         <dependency>
             <groupId>org.jenkins-ci.plugins</groupId>
             <artifactId>structs</artifactId>
-            <version>1.6</version>
+            <version>1.7-20170525.172515-3</version> <!-- TODO: Switch to release after https://github.com/jenkinsci/structs-plugin/pull/19 is merged and released -->
```
Comment (jglick) [inline pom.xml]: 1.7

**Comment by jglick** [comment]
Feel free to merge if CI build passes.

**Pull request merged**